dao/src/main/java/org/thingsboard/server/dao/sql/alarm/AlarmRepository.java

```diff
@@ -404,4 +392,25 @@ String updateAlarm(@Param("t_id") UUID tenantId, @Param("a_id") UUID alarmId, @P
     @Query(value = "DELETE FROM alarm_types AS at WHERE NOT EXISTS (SELECT 1 FROM alarm AS a WHERE a.tenant_id = at.tenant_id AND a.type = at.type) AND at.tenant_id = :tenantId AND at.type IN (:types)", nativeQuery = true)
     int deleteTypeIfNoAlarmsExist(@Param("tenantId") UUID tenantId, @Param("types") Set<String> types);
 
+    @Query(value = "SELECT a.id " +
+            "FROM AlarmEntity a " +
+            "WHERE a.tenantId = :tenantId " +
+            "AND a.originatorId = :originatorId " +
+            "AND ((:alarmTypes) IS NULL OR a.type IN (:alarmTypes)) " +
+            "AND ((:alarmSeverities) IS NULL OR a.severity IN (:alarmSeverities)) " +
+            "AND (a.cleared = false)",
+            countQuery = "" +
```

Comment:
Do we use count? Using Page will cause Spring Data to perform two queries - one for data and one for count.
Let's use `Slice` to avoid this (see findAlarmIdsByAssigneeId in this class)